src/helper/main.cpp

```diff
@@ -54,8 +55,10 @@ int main(int argc, char *argv[]) {
     auto logger = spdlog::stderr_color_mt("ddappsec");
     spdlog::set_default_logger(logger);
     logger->set_pattern("[%Y-%m-%d %H:%M:%S.%e][%l][%t] %v");
-    spdlog::set_level(
-        spdlog::level::from_str(config.get<std::string>("log_level")));
+
+    auto level = spdlog::level::from_str(config.get<std::string>("log_level"));
+    spdlog::set_level(level);
+    dds::waf::initialise_logging(level);
```

Comment:
If you look at previous commits I was using a static lambda to initialise the logger on the instance constructor, since there could be multiple instances of the WAF but the logger is global. However this made it rather tricky to test and I had to abandon the idea in favour of something more mundane. Any specific suggestion?

Reply:
You could still have this method exposed for logging but have it called lazily when the ddwaf is first used. But anyway, if this causes other problems (e.g. concurrency problems), then let it be. If other subscribers in the future requires this kind of one-off initialization we may implement a dedicated mechanism for it.